src/ert/_c_wrappers/enkf/runpaths.py

```diff
@@ -32,21 +34,26 @@ def __init__(
         job_name_format: str,
         runpath_format: str,
         filename: PathLike = Path(".ert_runpath_list"),
-        substitute: Callable[[str, int, int], str] = lambda x, *_: x,
+        context: Optional[SubstitutionList] = None,
```

Comment:
I preferred the existing interface of `Runpaths` where it did not directly depend on SubstitutionList. As `Runpaths` was defined, it should suffice to pass it `context.substitute_real_iter`.

Reply:
Passing callable functions defined internally in objects as part of constructor arguments does not really decouple the two objects that much, but that is also a working alternative I will change back.